File: python/stack_sorter.py

class StackSorter:
   

    def sort_stack(self, main_stack: list) -> None:
       
        aux_stack = []

        while main_stack:
            current_element = main_stack.pop()

            while aux_stack and aux_stack[-1] > current_element:
                main_stack.append(aux_stack.pop())

          
            aux_stack.append(current_element)

        
        while aux_stack:
            main_stack.append(aux_stack.pop())

File: src/utils/StackSorter.java
package utils;

import java.util.Stack;

public class StackSorter {

    public void sortStack(Stack<Integer> mainStack) {
        Stack<Integer> auxStack = new Stack<>();

        while (!mainStack.isEmpty()) {
            // 1 Sacar el elemento actual de la pila principal
            int currentElement = mainStack.pop();

            while (!auxStack.isEmpty() && auxStack.peek() > currentElement) {
                mainStack.push(auxStack.pop());
            }

            auxStack.push(currentElement);
        }

        while (!auxStack.isEmpty()) {
            mainStack.push(auxStack.pop());
        }
    }
}